packages/xstate-react/test/useActor.test.tsx

```diff
@@ -278,4 +278,64 @@ describe('useActor', () => {
     fireEvent.click(button);
     expect(div.textContent).toEqual('100');
   });
+
+  it('send() should be stable', (done) => {
+    jest.useFakeTimers();
+    const fakeSubscribe = () => {
+      return {
+        unsubscribe: () => {
+          /* ... */
+        }
+      };
+    };
+    const noop = () => {
+      /* ... */
+    };
+    const firstActor: ActorRefLike<any> = {
+      send: noop,
+      subscribe: fakeSubscribe
+    };
+    const lastActor: ActorRefLike<any> = {
+      send: () => {
+        done();
+      },
+      subscribe: fakeSubscribe
+    };
+
+    const Test = () => {
+      const [actor, setActor] = useState(firstActor);
+      const [, send] = useActor(actor);
+
+      React.useEffect(() => {
+        setTimeout(() => {
+          // The `send` here is closed-in
+          send('anything');
+        }, 10);
+      }, []); // Intentionally omit `send` from dependency array
+
+      return (
+        <>
+          <button
+            data-testid="button"
+            onClick={() => setActor(lastActor)}
+          ></button>
+        </>
+      );
+    };
+
+    const { getByTestId } = render(<Test />);
+
+    // At this point, `send` refers to the first (noop) actor
+
+    const button = getByTestId('button');
+    fireEvent.click(button);
+
+    // At this point, `send` refers to the last actor
+
+    jest.advanceTimersByTime(20);
+
+    // The effect will call the closed-in `send`, which originally
+    // was the reference to the first actor. Now that `send` is stable,
+    // it will always refer to the latest actor.
```

Comment:
I'm not super sure that this is the most correct logic - I mean, this is usually not how React works. I'm fine with it right now, but we might revisit this in the future maybe. 

I believe that the real issue reported on Spectrum was not about `send` not being stable when swapping actors but rather because currently, a single actor doesn't have a stable `send` which is related to the logic around deferred actors and this:
https://github.com/davidkpiano/xstate/blob/0165825e7e08407c9c5c8ab069fd5ffdd160242c/packages/xstate-react/src/useActor.ts#L20-L24

So while this PR solves that by just always providing the same `send` from the given `useActor` it kinda feels like a band-aid - fixing the problem a step further than it should be fixed. If we should have a stable `send` for a given `useActor` feels like a separate discussion to the original problem from Spectrum.